app/code/Magento/Shipping/view/adminhtml/templates/view/items.phtml

```diff
@@ -13,15 +13,22 @@
             </tr>
         </thead>
         <?php $_items = $block->getShipment()->getAllItems() ?>
-        <?php $_i = 0; foreach ($_items as $_item) :
-            if (!empty($_item->getOrderItem())) :
-                if ($_item->getOrderItem()->getParentItem()) :
+        <?php $_i = 0; foreach ($_items as $_item):
+            if (!empty($_item->getOrderItem())):
+                if ($_item->getOrderItem()->getParentItem()):
                     continue;
                 endif;
                 $_i++ ?>
                 <tbody class="<?= /* @noEscape */ $_i%2 ? 'odd' : 'even' ?>">
                     <?= $block->getItemHtml($_item) ?>
-                    <?= $block->getItemExtraInfoHtml($_item->getOrderItem()) ?>
+                    <?php $orderItem = $_item->getOrderItem(); ?>
+                    <?= $block->getItemExtraInfoHtml($orderItem) ?>
+                    <?php $qtyToShip = $orderItem->getQtyShipped() ??
+                        (float) $orderItem->getQtyOrdered() - (float) $orderItem->getQtyShipped(); ?>
+                    <?php if ($orderItem->getQtyShipped()): ?>
+                        <input type="hidden" name="shipment[items][<?= /** @noEscape */ $_item->getId()?>]"
+                               value="<?= /** @noEscape */ $qtyToShip; ?>">
+                    <?php endif; ?>
```

Comment:
I see that value from this input will be overridden to correct value from shipment, but proposed solution is confusing.
I believe that for hidden input value should be used `$_item->getQty()`

Reply:
> I see that value from this input will be overridden to correct value from shipment, but proposed solution is confusing. I believe that for hidden input value should be used `$_item->getQty()`

the issue is that if there is a partial shipment created on the order, we have to pass the remaining quatity to ship, not the entire item qty. if we use ```$_item->getQty()``` the entire order item qty will be passed, even if some of the qty have already been shipped.

Reply:
@johnulecan I'm not agree with you. For exist shipment label should be created for items from current shipment.
Also you may check https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Shipping/view/adminhtml/web/order/packaging.js#L571. Value from input will be replaced by item qty value if value from input is greater